### ISSUE TITLE
BOINC network/cpu times

### ISSUE THREAD

**Issue by romw** [BOINC/boinc-ng#591]
**Reported by Ageless on 3 Apr 38270819 14:40 UTC**
As reported by Markj at [the BOINC Dev forums](http://boinc.berkeley.edu/dev/forum_thread.php?id=2590):[[br]]

```
When I set the "network usage allowed" times they don't seem to work with certain
combinations.

In boingmgr -> advanced -> preferences and click on the network usage tab (also does
this for processor usage). Select time examples as below, click the okay button.
Reselect and they have changed.

06:20/22:20 becomes 06:19/22:19.
06:35/22:35 becomes 06:34/22:34.
06:50/22:50 becomes 06:49/22:49.

The times are stored in decimal, when they are converted back to a time they are not
shown correctly.

06:20 is stored as 6.333333 and when it converts from decimal back to a time it 
stuffs them up (ie .333333 x 60 mins = 19.999998) so it shows as 06:19.

See global_pref_override.xml exerpt below. I have seen this using BOINC versions 
5.10.30 and 5.10.45 under Windows.

<global_preferences>
<run_on_batteries>0</run_on_batteries>
<run_if_user_active>0</run_if_user_active>
<idle_time_to_run>3.000000</idle_time_to_run>
<start_hour>0.000000</start_hour>
<end_hour>0.000000</end_hour>
<net_start_hour>6.333333</net_start_hour>
<net_end_hour>22.333333</net_end_hour>
```

**Comment by romw** [comment]
**Commented by Nicolas on 28 Jun 38271558 14:40 UTC**
Ugh, it should be stored as hours*60+minutes. But I don't think that can be changed without breaking compatibility...

**Comment by romw** [comment]
**Commented by Nicolas on 10 Jul 38271610 05:46 UTC**
Replying to [Nicolas](comment:2):
> Ugh, it should be stored as hours*60+minutes.
Actually, I take that back. It should be stored as a ISO 8601 time, instead of a number. XML formats shouldn't be 1:1 mappings of the internal structures.

**Comment by romw** [comment]
**Commented by davea on 8 Sep 38271800 06:13 UTC**
Quick fix: round the display up to nearest second

**Comment by romw** [comment]
**Commented by Eric Myers on 4 Jul 38276615 15:33 UTC**
Times should be stored as seconds!

**Comment by romw** [comment]
**Commented by davea on 16 Jan 39161423 18:40 UTC**
(In [17409]) - manager: fix roundoff error in Advanced Prefs; fixes #613

**Comment by romw** [comment]
**Commented by romw on 9 Jul 39173072 00:53 UTC**
(In [17475]) 	- manager: fix roundoff error in Advanced Prefs; fixes #613

	clientgui/
		DlgAdvPreferences.cpp
